controllers/code_lookup.go

```diff
@@ -0,0 +1,50 @@
+package controllers
+
+import (
+	"encoding/json"
+	"github.com/intervention-engine/fhir/server"
+	"github.com/intervention-engine/ie/utilities"
+	"gopkg.in/mgo.v2/bson"
+	"net/http"
+)
+
+type code_request_form struct {
+	CodeSystem  string `json:"codesystem"`
+	Query       string `json:"query"`
+	ResultLimit int    `json:"limit"`
+}
+
+func CodeLookup(rw http.ResponseWriter, r *http.Request) {
+	decoder := json.NewDecoder(r.Body)
+
+	var codereqform code_request_form
+
+	err := decoder.Decode(&codereqform)
+	if err != nil {
+		http.Error(rw, err.Error(), http.StatusBadRequest)
+	}
+
+	codesystem, querystring := codereqform.CodeSystem, codereqform.Query
+	resultlimit := codereqform.ResultLimit
+
+	codecollection := server.Database.C("codelookup")
+
+	result := []utilities.CodeEntry{}
+
+	iter := codecollection.Find(bson.M{
+		"codeSystem": codesystem,
+		"$or": []interface{}{
+			bson.M{"code": bson.RegEx{Pattern: ".*" + querystring + ".*", Options: "i"}},
+			bson.M{"name": bson.RegEx{Pattern: ".*" + querystring + ".*", Options: "i"}},
+		}}).Limit(resultlimit).Iter()
+
+	err = iter.All(&result)
```

Comment:
Also nit-picky, but this intermediate iterator isn't necessary.  `Query` has an `All` function, so you don't have to make an `Iter` first.  If you like having it separated over two statements, you can still do this:

```go
query := codecollection.Find(bson.M{
	"codeSystem": codesystem,
	"$or": []interface{}{
		bson.M{"code": bson.RegEx{Pattern: ".*" + querystring + ".*", Options: "i"}},
		bson.M{"name": bson.RegEx{Pattern: ".*" + querystring + ".*", Options: "i"}},
	}}).Limit(resultlimit)

err = query.All(&result)
```

Or you can consolidate all to one statement:

```go
err = codecollection.Find(bson.M{
	"codeSystem": codesystem,
	"$or": []interface{}{
		bson.M{"code": bson.RegEx{Pattern: ".*" + querystring + ".*", Options: "i"}},
		bson.M{"name": bson.RegEx{Pattern: ".*" + querystring + ".*", Options: "i"}},
	}}).Limit(resultlimit).All(&result)
```